Tests/RingBufferTestsKiwi.m

```diff
@@ -14,113 +15,113 @@ @interface _RPTRingBuffer ()
-            });
-
-            it(@"should return a nil measurement when the passed index is equal to the buffer's size", ^{
-                _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:size];
-                [[measurement should] beNil];
-            });
+        it(@"should return a nil measurement when the passed index is equal to the buffer's size", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+
+            _RPTMeasurement *measurement = [buffer measurementAtIndex:10];
+
+            [[measurement should] beNil];
+        });
+    });
+
+    describe(@"measurementWithTrackingIdentifier", ^{
+
+        it(@"should return a non-nil measurement when the passed tracking identifier is in the filled range", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+            mkFillBuffer(buffer, 5, 9);
+
+            _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:6];
+
+            [[measurement should] beNonNil];
+        });
+
+        it(@"should return a measurement having the same tracking identifier as expected when the passed tracking identifier is in the filled range", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+            mkFillBuffer(buffer, 5, 9);
+
+            _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:6];
+
+            [[theValue(measurement.trackingIdentifier) should] equal:theValue(6)];
+        });
+
+        it(@"should return a nil measurement when the passed tracking identifier is not in the filled range", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+            mkFillBuffer(buffer, 5, 9);
+
+            _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:2];
+
+            [[measurement should] beNil];
         });
     });
 
     describe(@"nextMeasurement", ^{
 
         it(@"should return a non-nil measurement when the buffer is not full", ^{
             _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
-            _RPTMeasurement *measurement = nil;
-            for (int i = 0; i < 5; i++) {
-                measurement = [buffer nextMeasurement];
-            }
+            mkFillBuffer(buffer, 5, 10);
+
+            _RPTMeasurement *measurement = [buffer nextMeasurement];
+
             [[measurement should] beNonNil];
-            [[theValue(measurement.trackingIdentifier) should] equal:theValue(5)];
+        });
+
+        it(@"should return a non-nil measurement having expected tracking identifier when the buffer is not full", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+            mkFillBuffer(buffer, 5, 10);
+
+            _RPTMeasurement *measurement = [buffer nextMeasurement];
+
+            [[theValue(measurement.trackingIdentifier) should] equal:theValue(11)];
```

Comment:
Can you think of a way to make it clearer why we expect 11 here?

Reply:
the buffer is filled by the numbers from 1 to 10, so the next one should be 11.

Reply:
Yeah I know that but only cos I know how the implementation works. Maybe add a comment like
`// Tracking identifiers 1-10 have already been used so we expect the next available identifier to be 11` above the assert